Add initial implementation

nuragic opened the pull request:
Close #3.

salmanm reviewed (commented):
I tried it out and looks like it's not working. I see the text `Has linked issues? ` without the count as expected. In fact, it's even not showing up in this PRs checks even though an issue is linked.
Do you wanna describe how you've tested this action? 

Also, let's please add the [usual CI stuff](https://github.com/nearform/optic-release-automation-action/tree/main/.github/workflows)
- lint/test
- notify release
- optic automated release

nuragic commented:
@salmanm

> I tried it out and looks like it's not working. I see the text `Has linked issues? ` **without** the count. In fact, it's even not showing up in this PRs checks even though an issue is linked. Do you wanna describe how you've tested this action?

Great catch, thanks! 😅  I tested the action here with the `check.yml` workflow before adding tests and it worked; then I added tests but I've also split the code between `index.js` and `action.js` so the action wasn't actually running now because of the wrong path on the `build` script... fixed now!

> Also, let's please add the [usual CI stuff](https://github.com/nearform/optic-release-automation-action/tree/main/.github/workflows)
> 
> * lint/test
> * notify release
> * optic automated release

Added.

Eomm commented on src/action.js:
```diff
@@ -0,0 +1,75 @@
+    core.setOutput("linked_issues_count", linkedIssuesCount);
+
+    if (!linkedIssuesCount) {
+      core.setFailed(
```
Just to be sure: do we want to fail if there are zero linked issues?
This will lead to red CI if I'm not wrong, so I would mention the [`continue-on-error`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) setting into the readme

salmanm replied:
Yea good point, perhaps make it configurable? 
I was thinking that the action should post a comment if it finds no linked issues and `setFailed` should be called based on a configurable input (something like `mode: enforce|notify` idk)

simoneb replied:
@salmanm to get this out I think we can go with a simplified implementation:

- let's decide that the default behavior is to fail the build for now
- let's create an issue to make this behavior configurable. In the issue let's also include that we may want to consider implementing this behavior via [check runs](https://docs.github.com/en/rest/reference/checks) (I don't know if it really makes sense but just to research it really)

simoneb replied:
> Just to be sure: do we want to fail if there are zero linked issues? This will lead to red CI if I'm not wrong, so I would mention the [`continue-on-error`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) setting into the readme

I am not sure if that is the effect. I mean this action will need to be used in a standalone workflow because it needs some specific triggers. So failing it will only fail that workflow. If the workflow is not configured as required in the repo settings (which depends on the user to configure), then the overall build will not fail, only this workflow, which will prevent the PR from being merged only if the owner of the repo configured it as a required check (which it isn't by default).

I agree we can document this, but it's basic github knowledge and will not require any "continue-on-error"

salmanm reviewed (commented):
It's working fine now, let's just wait for the decision on [this thread](https://github.com/nearform/github-action-check-linked-issues/pull/1#discussion_r779048382).

Also, may be I'm missing something but I don't see when and how `dist/index.js` will be used! I mean when I reference this action `uses: nearform/github-action-check-linked-issues@main`, it would simply use the `main` file as per `package.json` which is `src/index.js`, right?

simoneb commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,34 @@
+        with:
+          node-version: 12
+
+      - uses: nearform/optic-release-automation-action@main
```
it's enabled on all repos in the nf org
